src/modules/Dashboard/features/RespondentData/RespondentDataSummary/hooks/useRespondentAnswers/useRespondentAnswers.ts

```diff
@@ -21,62 +21,92 @@ export const useRespondentAnswers = () => {
     startTime: providedStartTime,
     endTime: providedEndTime,
     filterByIdentifier: providedFilterByIdentifier,
+    identifier: providedIdentifier,
     versions: providedVersions,
+    isIdentifiersChange = false,
   }: FetchAnswers) => {
     if (!appletId || !respondentId) return;
+
     try {
       const {
-        startDate,
+        startDate: formStartDate,
         endDate: formEndDate,
         startTime: formStartTime,
         endTime: formEndTime,
-        identifier,
+        identifier: formIdentifier,
         filterByIdentifier: formFilterByIdentifier,
         versions: formVersions,
         identifiers,
       } = getValues();
-      const endDate = providedEndDate ?? formEndDate;
       const startTime = providedStartTime ?? formStartTime;
       const endTime = providedEndTime ?? formEndTime;
       const filterByIdentifier = providedFilterByIdentifier ?? formFilterByIdentifier;
       const versions = providedVersions ?? formVersions;
+      const identifier = providedIdentifier ?? formIdentifier;
+
+      const { selectedIdentifiers, recentAnswerDateString } =
+        getIdentifiers({
+          filterByIdentifier,
+          filterIdentifiers: identifier,
+          identifiers,
+        }) ?? {};
+
+      const {
+        identifierAnswerStartDate,
+        identifierAnswerEndDate,
+        activityAnswerStartDate,
+        activityAnswerEndDate,
+      } =
+        processIdentifiersChange({
+          setValue,
+          isIdentifiersChange,
+          adjustStartEndDates: !!(filterByIdentifier && identifier?.length),
+          recentIdentifiersAnswerDate: recentAnswerDateString,
+          activityLastAnswerDate: activity.lastAnswerDate,
+        }) ?? {};
 
-      const selectedIdentifiers = getIdentifiers(filterByIdentifier, identifier, identifiers);
+      const startDate = identifierAnswerStartDate ?? activityAnswerStartDate ?? formStartDate;
```

Comment:
I'm curious it's possibly missed one more fallback value `providedStartTime` like the same logic as it used for `endDate`?
```suggestion
      const startDate = identifierAnswerStartDate ?? activityAnswerStartDate ?? providedStartTime ?? formStartDate;
```

Reply:
No, this is not how it is expected. There is no ``providedStartDate`` in the properties.
![image](https://github.com/ChildMindInstitute/mindlogger-admin/assets/116339462/f085703c-699f-41f6-8a41-0ff3957c7262)